server/graphql/programs/src/lib.rs

```diff
@@ -110,6 +117,33 @@ impl ProgramsQueries {
         patient_search(ctx, store_id, input)
     }
 
+    pub async fn central_patient_search(
+        &self,
+        ctx: &Context<'_>,
+        store_id: String,
+        input: CentralPatientSearchInput,
+    ) -> Result<CentralPatientSearchResponse> {
+        // Note, we can't move the ctx to another async method because then it would need to be
+        // Sync. For this reason split the method as done below.
+        validate_auth(
+            ctx,
+            &ResourceAccessRequest {
+                resource: Resource::QueryPatient,
+                store_id: Some(store_id.clone()),
+            },
+        )?;
+
+        let service_provider = ctx.service_provider();
+        let context = service_provider.basic_context()?;
+
+        let sync_settings = service_provider.settings.sync_settings(&context)?.ok_or(
```

Comment:
Oh, yes this is a bit annoying. The reason for it is the `async` api call. The database connection is `Send` but not `Sync`. Which means the future for this whole method is not `Send` because the reference is hold across an await point.

This is the workaround to do the async api call (didn't want to block since the api call can take much longer than a DB request.

I think another workaround would be to move the `context` to a service function and then return it to the caller (just in case it is needed later), e.g.:
self
```rust
async fn service_call(context: ServiceContext) -> ServiceContext {
   // put same code from graphql here
}
```
Think I will actually try this, while it's different to how we pass the context elsewhere its still better than having this weird mix in the graphql endpoint.

Btw, there was a similar problem for the report data querying endpoint where we did an async API call...

There was also the same problem when trying async service calls ages ago. Back then the only solution I came up with was to put the connection into a mutex (which was only needed to make Rust happy and thus annoyed me). So moving (or sending) the connection around would be an alternative.

A probably even better alternative is to use this nightly Exclusive type which solves exactly the same problem:
https://doc.rust-lang.org/std/sync/struct.Exclusive.html
(This works by making a `Send + !Sync` type `Sync` by ensuring there is no shared access)

Reply:
Fix this now by moving the ServiceContext to the service.

Reply:
Ohhh, cool will check it out tomorrow thanks !

I still reckon we should consider moving to Arc<Mutex> connection: https://github.com/openmsupply/open-msupply/issues/78